test/middleware/request.js

```diff
@@ -102,19 +122,31 @@ test.cb('should be able to use provided authorization headers', (t) => {
     [CALL_API]: requestAction
   };
 
-  const actionHandler = t.context.nextHandler(action => {
-    t.deepEqual(action.config.headers.Authorization, 'Bearer another-token');
-    t.end();
+  const requestPromiseStub = sinon.stub().resolves({
+    body: {},
+    statusCode: 200
   });
+  const revertRequestStub = request.__set__('requestPromise', requestPromiseStub);
+
+  try {
+    const { invokeMiddleware } = create();
+
+    await invokeMiddleware(actionObj);
 
-  actionHandler(actionObj);
+    const nextAction = requestPromiseStub.firstCall.args[0];
+    t.deepEqual(nextAction.headers.Authorization, 'Bearer another-token');
+  } finally {
+    revertRequestStub();
+  }
 });
 
-test.cb('should dispatch error action for failed request', (t) => {
+test.serial('should dispatch error action for failed request', async (t) => {
   nock(`http://localhost:${port}`)
```

Comment:
I don't know why that was all PORT'd up, but you can make all of the tests use `http://anyhost`?